src/loaders/anndata-loaders/MatrixZarrLoader.js

```diff
@@ -31,25 +31,29 @@ const concatenateColumnVectors = (arr) => {
 /**
  * Loader for converting zarr into the a cell x gene matrix for use in Genes/Heatmap components.
  */
-export default class MatrixZarrLoader extends AbstractTwoStepLoader {
+export default class MatrixZarrLoader extends AnnDataLoader {
   /**
-   * Class method for loading the genes list from AnnData.var.
+   * Class method for loading the genes list from AnnData.var,
+   * filtered if a there is a `geneFilterZarr` present in the view config.
    * @returns {Promise} A promise for the zarr array contianing the gene names.
    */
-  async loadGeneNames() {
-    if (this.geneNames) {
-      return this.geneNames;
+  async loadFilteredGeneNames() {
+    if (this.filteredGeneNames) {
+      return this.filteredGeneNames;
     }
     const { geneFilter: geneFilterZarr } = this.options;
     let geneFilter;
     if (geneFilterZarr) {
       geneFilter = await this.dataSource.getFlatArrDecompressed(geneFilterZarr);
     }
-    const { _index } = await this.dataSource.getJson('var/.zattrs');
-    if (this.geneNames) {
-      return this.geneNames;
+    const geneNames = this.loadGeneNames();
+    if (this.filteredGeneNames) {
+      return this.filteredGeneNames;
     }
-    this.geneNames = this.dataSource.getFlatArrDecompressed(`var/${_index}`).then(data => data.filter((_, j) => !geneFilter || geneFilter[j])); return this.geneNames;
+    this.filteredGeneNames = geneNames.then(
+      data => data.filter((_, j) => !geneFilter || geneFilter[j]),
+    );
+    return this.filteredGeneNames;
```

Comment:
Thinking we could remove the `if` block above and just make this:

```javascript
	this.filteredGeneNames = this.loadGeneNames().then(
      data => data.filter((_, j) => !geneFilter || geneFilter[j]),
    );
    return this.filteredGeneNames;
```

Reply:
Unless I am missing something ...

Reply:
Good call.